src/txmempool.h

```diff
@@ -261,27 +261,44 @@ class CompareTxMemPoolEntryByEntryTime
     }
 };
 
+/** \class CompareTxMemPoolEntryByAncestorScore
+ *
+ *  Sort an entry by min(score/size of entry's tx, score/size with all ancestors).
+ */
 class CompareTxMemPoolEntryByAncestorFee
 {
 public:
     bool operator()(const CTxMemPoolEntry& a, const CTxMemPoolEntry& b) const
     {
-        double aFees = a.GetModFeesWithAncestors();
-        double aSize = a.GetSizeWithAncestors();
+        bool fUseAAncestors = UseAncestorScore(a);
+        bool fUseBAncestors = UseAncestorScore(b);
+
+        double aModFee = fUseAAncestors ? a.GetModFeesWithAncestors() : a.GetModifiedFee();
+        double aSize = fUseAAncestors ? a.GetSizeWithAncestors() : a.GetTxSize();
 
-        double bFees = b.GetModFeesWithAncestors();
-        double bSize = b.GetSizeWithAncestors();
+        double bModFee = fUseBAncestors ? b.GetModFeesWithAncestors() : b.GetModifiedFee();
```

Comment:
Instead of factoring out `UseAncestorScore`, do (missing comments):
```cpp
void GetModFeeAndSize(const CTxMemPoolEntry& a, int64_t& mod_fee, int64_t& size) {
    int64_t f1 = a.GetModifiedFee() * a.GetSizeWithAncestors();
    int64_t f2 = a.GetModFeesWithAncestors() * a.GetTxSize();
    if (f1 > f2) {
        mod_fee = a.GetModFeesWithAncestors();
        size = a.GetSizeWithAncestors();
    } else {
        mod_fee = a.GetModifiedFee();
        size = a.GetTxSize();
    }
}
```